lib/src/kbasesearchengine/main/DecorateWithNarrativeInfo.java

```diff
@@ -0,0 +1,121 @@
+                searchObjsOutput.getAccessGroupNarrativeInfo()));
+        return searchObjsOutput;
+    }
+
+    public GetObjectsOutput getObjects(GetObjectsInput params, String user)
+            throws Exception {
+        GetObjectsOutput getObjsOutput = searchInterface.getObjects(params, user);
+        getObjsOutput.setAccessGroupNarrativeInfo(addNarrativeInfo(getObjsOutput.getObjects(),
+                getObjsOutput.getAccessGroupNarrativeInfo()));
+        return getObjsOutput;
+    }
+
+    private Map<Long, Tuple5 <String, Long, String, String, String>> addNarrativeInfo(
+            List<ObjectData> objects,
+            Map<Long, Tuple5 <String, Long, String, String, String>> accessGroupNarrInfo)
+                throws IOException, JsonClientException {
+
+        long wsId;
+        Tuple9 <Long, String, String, String, Long, String, String,
+                String,Map<String,String>> wsInfo;
+        Tuple5 <String, Long, String, String, String> tempNarrInfo =
+                new Tuple5<>();
+        final Map<Long, Tuple5 <String, Long, String, String, String>> retVal = new HashMap<>();
+
+        if (accessGroupNarrInfo != null) {
+            retVal.putAll(accessGroupNarrInfo);
+        }
+
+        List<Long> wsIdsList = new ArrayList<Long>();
+
+        for (ObjectData objData: objects) {
+
+            GUID guid = new GUID(objData.getGuid());
+
+            String storageCode = guid.getStorageCode();
+
+            if (STORAGE_CODE.equals(storageCode)) {
+                wsId = guid.getAccessGroupId();
+                if (!wsIdsList.contains(wsId))
+                    wsIdsList.add(wsId);
+            }
+            else {
+                // TODO: handle other storage types?
+            }
+        }
+        for (long workspaceId: wsIdsList) {
+            if (!retVal.containsKey(workspaceId)) {
+                // get workspace info meta data
+                WorkspaceEventHandler weh = new WorkspaceEventHandler(
+                                                new CloneableWorkspaceClientImpl(wsClient));
+                wsInfo = weh.getWorkspaceInfo(workspaceId);
+                tempNarrInfo.setE3(wsInfo.getE4());
+                tempNarrInfo.setE4(wsInfo.getE3());
+                tempNarrInfo.setE5("TBD");
+
+                Map<String, String> wsInfoMeta = wsInfo.getE9();
+
+                if ( wsInfoMeta.containsKey("narrative") &&
+                        wsInfoMeta.containsKey("narrative_nice_name") ) {
+                    tempNarrInfo.setE1(wsInfoMeta.get("narrative_nice_name"));
+                    tempNarrInfo.setE2(Long.parseLong(wsInfoMeta.get("narrative")));
+                }
+                else { // will get here only when testing locally
+                    tempNarrInfo.setE1("Not Available");
+                    tempNarrInfo.setE2((long)-1);
+                }
+                retVal.put(workspaceId, tempNarrInfo);
```

Comment:
This is making multiple references to the same tempNarrInfo, so all the workspaces will be shown as having the same info, unless I'm misunderstanding something.

Reply:
I have changed and tested this part with more than one workspace in the results. Found to be working.